multiversx_sdk/network_providers/transaction_awaiter.py

```diff
@@ -92,6 +97,8 @@ def _await_conditionally(self,
 
                 if is_condition_satisfied:
                     break
+            except TransactionFetchingError:
```

Comment:
It would work in this case, but this error it's also used in the network providers. I would lean towards not changing the name, but if you say we should, I will.